### ISSUE TITLE
AWS API gateway timeout makes config timeout irrelevant

### ISSUE THREAD

**Issue by tomaszps** [aws/chalice#1580]
Hello,

The AWS API gateway has a hard, non-configurable timeout at 30s. I couldn't find in the docs whether *all* chalice deployments use it, but at least several of the standard deployments do. That makes the parameter in your [config](https://aws.github.io/chalice/topics/configfile.html?highlight=time):
`lambda_timeout`
irrelevant, I believe?

I'm not asking for a workaround at this point; I've abandoned this approach. I just want to change the docs to reflect this reality by mentioning the API gateway timeout, which is pretty non-obvious.

Thanks!

**Comment by jamesls** [comment]
That's a good point about the docs, I'll get that updated.  We should document and potentially warn/validate if you set a timeout that's larger than the API gateway timeout.  However, you can also use Chalice without the REST API portion (e.g. event handlers to SQS, SNS, DynamoDB, etc.) in which case the lambda timeout's are still useful.

**Comment by adamlukaszewski** [comment]
That's becomes a big issue when the lambda function stands behind the api gateway. We have some lambdas which may work for 1 - 3 minutes. They will be triggered via api gateway. Currently we are facing the problem that the api gateway responses with 504 (gateway timed out). 

We need an possibility to change not only the timeout of the lambda function itself, but also the timeout of the api gateway. 

So it is not only about documentation, folks.

**Comment by tomaszps** [comment]
@adamlukaszewski I believe you're barking up the wrong tree here, from what I fuzzily remember of this issue. It's not something chalice can change- it's an AWS limit on this service that can't be changed via an API call.

**Comment by adamlukaszewski** [comment]
Yeah see it, now. Thanks for clarification. Here the link to the doc:
https://docs.aws.amazon.com/apigateway/latest/developerguide/limits.html
